packages/main/src/plugin/proxy.ts

```diff
@@ -104,58 +109,76 @@ export class Proxy {
     this.configurationRegistry.registerConfigurations([proxyConfigurationNode]);
 
     // be notified when configuration is updated
-    this.configurationRegistry.onDidChangeConfiguration(e => {
+    this.configurationRegistry.onDidChangeConfiguration(async e => {
       if (e.key.startsWith('proxy')) {
-        this.updateFromConfiguration();
+        await this.updateFromConfiguration();
       }
     });
 
     // read initial value
-    this.updateFromConfiguration();
+    await this.updateFromConfiguration();
     this.overrideFetch();
   }
 
-  updateFromConfiguration(): void {
+  async updateFromConfiguration(): Promise<void> {
     // read value from the configuration
     const proxyConfiguration = this.configurationRegistry.getConfiguration('proxy');
-    const httpProxy = proxyConfiguration.get<string>('http');
-    const httpsProxy = proxyConfiguration.get<string>('https');
-    const noProxy = proxyConfiguration.get<string>('no');
-    const isEnabled = proxyConfiguration.get<boolean>('enabled') ?? false;
-
-    this.proxySettings = {
-      httpProxy: ensureURL(httpProxy),
-      httpsProxy: ensureURL(httpsProxy),
-      noProxy,
-    };
-    this.proxyState = isEnabled;
+    const isEnabled = proxyConfiguration.get<unknown>('enabled');
+    let proxyState: ProxyState = ProxyState.PROXY_SYSTEM;
+    if (typeof isEnabled === 'boolean') {
+      proxyState = isEnabled ? ProxyState.PROXY_MANUAL : ProxyState.PROXY_SYSTEM;
```

Comment:
Default to PROXY_SYSTEM will make podman desktop to fail to load after restart in corner cases like:
1. Running on unsupported platform
2. Detection fails on supported platform

While (1) should be unlikely, (2) will happen because Murphy Law.

Proxy.init() called in PluginSystem.initExtensions() outside of try catch and that will end with error below

```
UnhandledPromiseRejectionWarning: Error: Unsupported environment
    at getProxySettingsFromSystem (/Users/eskimo/Sources/podman-desktop/packages/main/dist/index.cjs:122228:9)
    at Proxy2.updateFromConfiguration (/Users/eskimo/Sources/podman-desktop/packages/main/dist/index.cjs:122326:34)
    at Proxy2.init (/Users/eskimo/Sources/podman-desktop/packages/main/dist/index.cjs:122303:16)
    at PluginSystem.initExtensions (/Users/eskimo/Sources/podman-desktop/packages/main/dist/index.cjs:127287:17)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async /Users/eskimo/Sources/podman-desktop/packages/main/dist/index.cjs:129959:23
```